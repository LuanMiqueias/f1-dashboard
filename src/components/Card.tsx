import { ReactNode } from "react";
import styles from "../styles/components/Card.module.css";
interface PropsCard {
  position: number;
  name: string;
  team: string;
  isPlayer?: boolean;
}
const Card = ({ name, position, team, isPlayer }: PropsCard) => {
  return (
    <div className={styles.card}>
      <p>
        <span className={styles.colorPosition}></span>
        {position}Â°
      </p>
      <div className={styles.content_card}>
        <h2>{name}</h2>
        <p>{team}</p>
      </div>
    </div>
  );
};

export default Card;
