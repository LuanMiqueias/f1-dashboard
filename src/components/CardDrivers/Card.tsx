import React, { ReactNode } from "react";
import Loading from "../Loading/Loading";
import styles from "./style.module.css";
interface PropsCard {
  position: number;
  name: string;
  team: string;
  image: string;
  points: number;
  isPlayer?: boolean;
}
const Card = ({ name, position, team, image, points, isPlayer }: PropsCard) => {
  const [showInfo, setShowInfo] = React.useState(false);
  const [imgLoading, setImgLoading] = React.useState(true);

  return (
    <>
      <div className={styles.card} onClick={() => setShowInfo(!showInfo)}>
        <p>
          <span className={styles.colorPosition}></span>
          {position}Â°
        </p>
        <div className={styles.content_card}>
          <h2>{name}</h2>
          <p>{team}</p>
        </div>
      </div>
      {showInfo && (
        <div className={styles.cardInfo}>
          <div className={styles.cardInfo_content}>
            <div className={styles.img_driver}>
              {imgLoading && (
                <div className={styles.loading_img}>
                  <Loading />
                </div>
              )}
              <img
                src={image}
                alt={name}
                className={!imgLoading ? styles.img_loaded : ""}
                onLoad={() => setImgLoading(false)}
              />
            </div>
            <div className={styles.cardInfo_content_text}>
              <p>
                <span>Nationality:</span>British
              </p>
              <p>
                <span>Points:</span>
                {points}
              </p>
              <p>
                <span>Team:</span>
                {team}
              </p>
            </div>
          </div>
        </div>
      )}
    </>
  );
};

export default Card;
