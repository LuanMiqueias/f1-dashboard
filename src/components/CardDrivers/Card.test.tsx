import { fireEvent, render, screen } from "@testing-library/react";
import Card from "./Card";

const fakeData = {
  position: 1,
  name: "name1",
  team: "team1",
  image: "img1",
  points: 10,
};
it("Should math snapshot", () => {
  const { container } = render(<Card {...fakeData} />);

  expect(container.firstChild).toMatchSnapshot();
});
it("Should have props if showInfo is equal to false", () => {
  render(<Card {...fakeData} />);

  //Name
  expect(screen.getByRole("heading", { name: /name1/i }));
  //Position
  expect(screen.getByText(/1Â°/i)).toBeInTheDocument();
  //Team
  expect(screen.getByText(/team1/i)).toBeInTheDocument();
  //Points
  expect(screen.queryByText(/10/i)).not.toBeInTheDocument();
  //Image Driver
  expect(screen.queryByRole("img", { name: /name1/i })).not.toBeInTheDocument();
});

it("Should have props if showInfo is equal to true", () => {
  const { container } = render(<Card {...fakeData} />);
  fireEvent.click(container.firstElementChild);

  //Image Driver
  expect(screen.getByRole("img", { name: /name1/i })).toBeInTheDocument();
  //Points
  expect(screen.getByText(/10/i)).toBeInTheDocument();
});
