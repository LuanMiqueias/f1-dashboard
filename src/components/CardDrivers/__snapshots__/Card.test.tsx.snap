// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="card"
>
  <p>
    <span
      class="colorPosition"
    />
    1
    Â°
  </p>
  <div
    class="content_card"
  >
    <h2>
      name1
    </h2>
    <p>
      team1
    </p>
  </div>
</div>
`;
